plugin/src/main/java/org/gradle/testretry/TestRetryPlugin.java

```diff
@@ -39,9 +39,16 @@
 
     @Override
     public void apply(Project project) {
+        if (pluginAlreadyApplied(project)) {
+            return;
+        }
+
         project.getTasks()
             .withType(Test.class)
             .configureEach(task -> configureTestTask(task, objectFactory, providerFactory));
     }
 
+    private static boolean pluginAlreadyApplied(Project project) {
```

Comment:
A matter of preference but how about 
```suggestion
    private static boolean isPluginAlreadyApplied(Project project) {
```